homeassistant/components/wiser/__init__.py

```diff
@@ -0,0 +1,362 @@
+Angelo.santagata@gmail.com
+"""
+import asyncio
+from datetime import timedelta
+from functools import partial
+import json
+
+import requests.exceptions
+import voluptuous as vol
+from wiserHeatingAPI.wiserHub import (
+    TEMP_MAXIMUM,
+    TEMP_MINIMUM,
+    WiserHubTimeoutException,
+    wiserHub,
+)
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MINIMUM,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import Throttle
+
+from .const import (
+    _LOGGER,
+    CONF_BOOST_TEMP,
+    CONF_BOOST_TEMP_TIME,
+    DATA,
+    DEFAULT_BOOST_TEMP,
+    DEFAULT_BOOST_TEMP_TIME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    HUBNAME,
+    MANUFACTURER,
+    UPDATE_LISTENER,
+    UPDATE_TRACK,
+    WISER_PLATFORMS,
+    WISER_SERVICES,
+)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            cv.ensure_list,
+            [
+                {
+                    vol.Required(CONF_HOST): cv.string,
+                    vol.Required(CONF_PASSWORD): cv.string,
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                    ): vol.All(vol.Coerce(int)),
+                    vol.Optional(CONF_MINIMUM, default=TEMP_MINIMUM): vol.All(
+                        vol.Coerce(int)
+                    ),
+                    vol.Optional(CONF_BOOST_TEMP, default=DEFAULT_BOOST_TEMP): vol.All(
+                        vol.Coerce(int)
+                    ),
+                    vol.Optional(
+                        CONF_BOOST_TEMP_TIME, default=DEFAULT_BOOST_TEMP_TIME
+                    ): vol.All(vol.Coerce(int)),
+                }
+            ],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """
+    Wiser uses config flow for configuration.
+
+    But, a "wiser:" entry in configuration.yaml will trigger an import flow
+    if a config entry doesn't already exist. If it exists, the import
+    flow will attempt to import it and create a config entry, to assist users
+    migrating from the old wiser component. Otherwise, the user will have to
+    continue setting up the integration via the config flow.
+    """
+    hass.data.setdefault(DOMAIN, {})
+
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf,
```

Comment:
This is a new integration and thus, no previous data to import.
Please remove this part/feature.

Reply:
done